specs/core/0_beacon-chain.md

```diff
@@ -1509,10 +1527,36 @@ For each `attestation` in `block.body.attestations`:
 * Verify that `attestation.data.justified_slot` is equal to `state.justified_slot if attestation.data.slot >= state.slot - (state.slot % EPOCH_LENGTH) else state.previous_justified_slot`.
 * Verify that `attestation.data.justified_block_root` is equal to `get_block_root(state, attestation.data.justified_slot)`.
 * Verify that either `attestation.data.latest_crosslink_root` or `attestation.data.shard_block_root` equals `state.latest_crosslinks[shard].shard_block_root`.
-* `aggregate_signature` verification:
-    * Let `participants = get_attestation_participants(state, attestation.data, attestation.aggregation_bitfield)`.
-    * Let `group_public_key = bls_aggregate_pubkeys([state.validator_registry[v].pubkey for v in participants])`.
-    * Verify that `bls_verify(pubkey=group_public_key, message=hash_tree_root(AttestationDataAndCustodyBit(attestation.data, False)), signature=attestation.aggregate_signature, domain=get_domain(state.fork, attestation.data.slot, DOMAIN_ATTESTATION))`.
+* Verify bitfields:
+
+```python
+    assert verify_bitfield(attestation.aggregation_bitfield)
+    assert verify_bitfield(attestation.custody_bitfield)
+    assert attestation.custody_bitfield & attestation.aggregation_bitfield == attestation.custody_bitfield
```

Comment:
Is that the right way of doing bitwise AND? @djrtwo

Reply:
`bytes` in python is a list of immutable ints. The operation would need to be performed on each position. I think the following is the functionality you are looking for but isn't pretty:

```python
and_bitfield = [
    custody_byte & aggregate_byte 
    for custody_byte, aggregate_byte in zip(attestation.custody_bitfield, attestation.aggregation_bitfield)
]
assert and_bitfield == attestation.custody_bitfield
```

Reply:
```python
for i in range(len(crosslink_committee)):
    if get_bitfield_bit(attestation.aggregation_bitfield) == 0:
        assert get_bitfield_bit(attestation.custody_bitfield) == 0
```

More verbose but it clarifies what we're trying to do much better imo.